code/lib/cli/src/migrate.ts

```diff
@@ -10,11 +10,14 @@ import { getStorybookVersionSpecifier } from './helpers';
 
 const logger = console;
 
-export async function migrate(migration: any, { glob, dryRun, list, rename, parser }: any) {
+export async function migrate(
+  migration: any,
+  { glob, dryRun, list, rename, parser, runAutomigration = true }: any
+) {
   if (list) {
     listCodemods().forEach((key: any) => logger.log(key));
   } else if (migration) {
-    if (migration === 'mdx-to-csf' && !dryRun) {
+    if (migration === 'mdx-to-csf' && !dryRun && runAutomigration) {
```

Comment:
I think you have to adjust line 41-46 (can't mark it), because:

If the user runs the codemod `mdx-to-csf`:
- It will run the automigration `bareMdxStoriesGlob` which will also run the codemod `mdx-to-csf`. If I am not wrong, the `mdx-to-csf` codemod would be applied twice.

So line 41-46 shouldn't call the `bareMdxStoriesGlob` automigration, but instead it should do:

```js
await updateMainConfig({ mainConfigPath, dryRun: !!dryRun }, async (main) => {
  main.setFieldValue(['stories'], nextStoriesEntries);
});
```

So essentially it executes the code here: https://github.com/storybookjs/storybook/pull/26201/files#diff-f3ca96c51596e98ed664f151ef17533610c472b0dbba6300ad7867053b1209d1R133-R135
Feel free to extract it, so that you can use it in the `bare-mdx-stories-glob` automigration and in the `mdx-to-csf` codemod.

Reply:
I don't know what to do here.
[The link](https://github.com/storybookjs/storybook/pull/26201/files#diff-f3ca96c51596e98ed664f151ef17533610c472b0dbba6300ad7867053b1209d1R133-R135) doesn't seem to go anywhere, for me.

I tried to follow the safest possible path with the least amount of changes, because we're in crunch-mode for RC?

I would rather not refactor code between codemods and CLI packages, right now.